File: backend/src/main/java/com/profroid/profroidapp/cellarsubdomain/presentationLayer/CellarResponseModel.java
package com.profroid.profroidapp.cellarsubdomain.presentationLayer;

import com.profroid.profroidapp.cellarsubdomain.dataAccessLayer.CellarType;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CellarResponseModel {

    private String cellarId;
    private String ownerCustomerId;

    private String name;

    private double height;
    private double width;
    private double depth;

    private Integer bottleCapacity;

    private boolean hasCoolingSystem;
    private boolean hasHumidityControl;
    private boolean hasAutoRegulation;

    private CellarType cellarType;
}

File: frontend/src/features/cellar/models/CellarResponseModel.ts
import type { CellarType } from "./CellarType";

export interface CellarResponseModel {
  cellarId: string;
  ownerCustomerId: string;

  name: string;

  height: number;
  width: number;
  depth: number;

  bottleCapacity: number;

  hasCoolingSystem: boolean;
  hasHumidityControl: boolean;
  hasAutoRegulation: boolean;

  cellarType: CellarType;
}
